src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -549,21 +551,26 @@ private void InitializeHub(THub hub, HubConnectionContext connection)
 
     private static Task<bool> IsHubMethodAuthorized(IServiceProvider provider, HubConnectionContext hubConnectionContext, HubMethodDescriptor descriptor, object?[] hubMethodArguments, Hub hub)
     {
+        var endpoint = hubConnectionContext.Features.Get<IEndpointFeature>()?.Endpoint;
```

Comment:
Move this to where HubMethodDescriptor gets its AuthorizeData, we don't want the cost of these calls per Hub method invoke since it's static information.